src/umpire/strategy/mixins/Inspector.cpp

```diff
@@ -12,41 +12,34 @@ namespace umpire {
 namespace strategy {
 namespace mixins {
 
-Inspector::Inspector() :
-  m_current_size(0),
-  m_high_watermark(0),
-  m_allocation_count{0}
-{
-}
-
 void
 Inspector::registerAllocation(
     void* ptr,
     std::size_t size,
-    strategy::AllocationStrategy* strategy)
+    strategy::AllocationStrategy* s) 
 {
-  m_current_size += size;
-  m_allocation_count++;
+  s->m_current_size += size;
```

Comment:
Directly changing state of another object seems sneaky to me.  I appreciate why it is being done this way, but is there perhaps a different way to accomplish the same thing without having to worry about the strategy object and inspector object from inadvertantly overwriting each other?